### ISSUE TITLE
Pyreverse not showing sub-classes with -c flag

### ISSUE THREAD

**Issue by fievelk** [pylint-dev/pylint#7361]
### Bug description

Directory tree:
```
pyreverse_inheritance
├── __init__.py
└── src
    ├── __init__.py
    └── normal_inheritance.py
```

Contents of `normal_inheritance.py`:
```py
# pylint: disable=missing-module-docstring, too-few-public-methods, missing-class-docstring
# pylint: disable=missing-function-docstring
class SomeParent:
    def __init__(self):
        self.some_variable = {}

    def get_some_variable(self):
        return "some-variable"


class SomeChild(SomeParent):
    def __init__(self):
        super().__init__()
        self.some_other_variable = {}
```

### Command used

```shell
pyreverse --only-classnames -c pyreverse_inheritance.src.normal_inheritance.SomeParent .
```

### Pylint output

```shell
$ pylint src

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

The docs state:
> --class <class>, -c <class>
                        create a class diagram with all classes related to \<class\>; this uses by default the options -ASmy (default: [])

So I would expect the output to show both classes (`SomeParent` and `SomeChild`) and their inheritance. Instead, only the parent class (`SomeParent`) is shown in the output.
If I remove the `-c` flag the inheritance is correctly shown.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.8.12 (9ef55f6fc369, Oct 25 2021, 05:10:01)
[PyPy 7.3.7 with GCC Apple LLVM 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

MacOS Monterey 12.5.1

**Comment by DudeNr33** [comment]
Thank you for opening this issue.

The options `-ASmy` stand for:

* `-A` (`--all-ancestors`): *show all ancestors off all classes in <projects>* --> Shows all *parent* classes
* `-S` (`--all-associated`): *show recursively all associated off all associated classes* --> Shows all classes that the our class has a *has-a* or *uses* relationship with
* `-my` (`--module-names=y`): *include module name in representation of classes*  --> irrelevant here

In that sense, `pyreverse` is doing exactly what the command line options are supposed to do. I am afraid that the feature of showing all *child* classes does not exist yet. 
If you need this feature, I would leave this ticket open, and I would be happy to review a pull request adding something like a `--all-derived` option.
